events/src/test-support/java/tech/pegasys/artemis/events/TestExceptionHandler.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.events;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import org.apache.logging.log4j.LogManager;
+
+public enum TestExceptionHandler implements ChannelExceptionHandler {
+  TEST_EXCEPTION_HANDLER;
+
+  @Override
+  public void handleException(
+      final Throwable error,
+      final Object subscriber,
+      final Method invokedMethod,
+      final Object[] args) {
+    LogManager.getLogger(invokedMethod.getClass())
+        .error(
+            "Unhandled error in subscriber "
+                + subscriber.getClass().getName()
+                + " for method "
+                + invokedMethod.getDeclaringClass().getName()
+                + "."
+                + invokedMethod.getName()
+                + " with arguments "
+                + Arrays.toString(args),
+            error);
```

Comment:
Rather than logging the error, we should just rethrow it to cause the test to fail (we're being called synchronously).  I'd suggest using `Assertions.fail("Unhandled error in subscriber ...", error)`.  That will give lots of context but also wind up throwing an `AssertionError` which will almost always bubble all the way up and fail the test (whereas `RuntimeException` might be caught).

Reply:
What if the test is failing on purpose, and the logging of some failure is expected? Wouldn't replacing the log with a fail() method break that?

Reply:
Seems unlikely, but let's cross that bridge when we get to it.  It's always better to default to blowing up in tests rather than logging and carrying on to let the test pass.